private/buf/cmd/buf/command/config/configlsmodules/configlsmodules.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configlsmodules
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"sort"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/bufpkg/bufconfig"
+	"github.com/bufbuild/buf/private/pkg/app"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appext"
+	"github.com/bufbuild/buf/private/pkg/stringutil"
+	"github.com/bufbuild/buf/private/pkg/syserror"
+	"github.com/spf13/pflag"
+)
+
+const (
+	configFlagName = "config"
+	formatFlagName = "format"
+
+	formatPath = "path"
+	formatName = "name"
+	formatJSON = "json"
+
+	defaultFormat = formatPath
+)
+
+var (
+	allFormats = []string{
+		formatPath,
+		formatName,
+		formatJSON,
+	}
+)
+
+// NewCommand returns a new Command.
+func NewCommand(
+	name string,
+	builder appext.SubCommandBuilder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name,
+		Short: "List configured modules",
+		Args:  appcmd.NoArgs,
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appext.Container) error {
+				return run(ctx, container, flags)
+			},
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct {
+	Config string
+	Format string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(
+		&f.Config,
+		configFlagName,
+		"",
+		`The buf.yaml file or data to use for configuration.`,
```

Comment:
The implementation first looks for a `buf.work.yaml` file then falls back to a `buf.yaml` file. Do we want the `--config` argument to accept either a `buf.yaml` or a `buf.work.yaml` file?

Reply:
I think that's a good point:
* by default, the command looks for `./buf.work.yaml` or `./buf.yaml`
* but the `--config` currently only accepts `buf.yaml`

It does feel a bit inconsistent. Do you have thoughts on this? @bufdev @doriable @emcfarlane